src/EntryPoints/WikibaseFacetedSearchHooks.php

```diff
@@ -26,11 +35,93 @@ public static function onShowSearchHitTitle(
 		array &$query,
 		array &$attributes
 	): void {
-		if ( $title->getNamespace() !== WB_NS_ITEM ) {
+		$itemId = self::getItemId( $title );
+
+		if ( $itemId === null ) {
+			return;
+		}
+
+		$pageTitle = self::getItemPage( $itemId );
+
+		if ( $pageTitle === null ) {
 			return;
 		}
 
-		// TODO: get item site link and replace $title
+		$title = $pageTitle;
+
+		if ( !( $result instanceof EntityResult ) ) {
+			self::rewriteLinkForNonEntityResult(
+				self::newLabelDescriptionLookup( $specialSearch->getContext() ),
+				self::newLinkFormatter( $specialSearch->getLanguage() ),
+				$itemId,
+				$titleSnippet,
+				$attributes
+			);
+		}
+	}
```

Comment:
For non-EntityResult there is no $titleSnippet by default, which means the link is going to use the text of the target page title, not the item title.

To get the item title again, this calls a minimal implementation based on `HtmlPageLinkRendererEndHookHandler::internalDoHtmlPageLinkRendererEnd()`.